### PR TITLE
give the postion-store thread its own connection

### PR DESCRIPTION
**Pull request by osheroff**
also, make sure we store the position one last time before bail

### CODE REVIEW

**Comment by osheroff** [comment]
@vanchi-zendesk

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 1]
```diff
@@ -39,11 +39,15 @@ public MaxwellConfig() {
...
+	private Connection newMasterConnection() throws SQLException {
+		return DriverManager.getConnection("jdbc:mysql://" + mysqlHost + ":" + mysqlPort, mysqlUser, mysqlPassword);
+	}
+
 	public Connection getMasterConnection() throws SQLException {
 		if ( this.connection != null )
 			return this.connection;
 
-		this.connection = DriverManager.getConnection("jdbc:mysql://" + mysqlHost + ":" + mysqlPort, mysqlUser, mysqlPassword);
+		this.connection = newMasterConnection();
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: There is no per-thread storage?

**Comment by vanchi-zendesk** [comment]
:+1:

**Comment by osheroff** [comment]
oh there, totally is thread-local storage, just didn't want to deal with a connection pool manager (yet).

**Pull request merged**